Updated GTStructure to use pointers to items instead of value receivers

jaroslaw-bochniak opened the pull request:
Updated GTStructure to use pointers to items instead of value receivers.
This is a breaking change as the public method signatures and main type has been modified.

- allows creation of the structure and flexible mutating it before actual printing
- pointer receiver should be more efficient for large data structures
- added simple test + test resources

jaroslaw-bochniak committed:
Updated GTStructure to use pointers to items instead of value receivers. (#1)

This is a breaking change as the public method signatures and main type has been modified.

- allows creation of the structure and flexible mutating it before actual printing
- pointer receiver should be more efficient for large structures
- added simple test + test resources

d6o commented on gotree_test.go:
```diff
@@ -0,0 +1,57 @@
+)
+
+// TestFolderTree should test if reading a folder returns rendered structure as expected
+func TestFolderTree(t *testing.T) {
```
I don't like the idea of keeping the testfiles in the project, we should Improve the ReadFolder to be mockable or remove this test for now.

jaroslaw-bochniak replied:
Sure no problem.

jaroslaw-bochniak committed:
After review improvements.

- removed test folder we should find a better solution for testing `ReadFolder` method
- updated readme to reflect signature changes
